lib/phoenix_live_view.ex

```diff
@@ -1842,4 +1966,99 @@ defmodule Phoenix.LiveView do
       |> Map.update!(:__changed__, &MapSet.put(&1, name))
     end)
   end
+
+  @doc """
+  Assigns keys asynchronously.
+
+  The task is linked to the caller and errors are wrapped.
+  Each key passed to `assign_async/3` will be assigned to
+  an `%AsyncResult{}` struct holding the status of the operation
+  and the result when completed.
+
+  ## Examples
+
+      def mount(%{"slug" => slug}, _, socket) do
+        {:ok,
+         socket
+         |> assign(:foo, "bar")
+         |> assign_async(:org, fn -> {:ok, %{org: fetch_org!(slug)} end)
+         |> assign_async([:profile, :rank], fn -> {:ok, %{profile: ..., rank: ...}} end)}
+      end
+
+  See the moduledoc for more information.
+  """
+  def assign_async(%Socket{} = socket, key_or_keys, func)
+      when (is_atom(key_or_keys) or is_list(key_or_keys)) and
+             is_function(func, 0) do
+    Async.assign_async(socket, key_or_keys, func)
+  end
+
+  @doc """
+  Starts an ansynchronous task and invokes callback to handle the result.
+
+  The task is linked to the caller and errors/exits are wrapped.
+  The result of the task is sent to the `handle_async/3` callback
+  of the caller LiveView or LiveComponent.
+
+  ## Examples
+
+      def mount(%{"id" => id}, _, socket) do
+        {:ok,
+         socket
+         |> assign(:org, AsyncResult.loading())
+         |> start_async(:my_task, fn -> fetch_org!(id) end)
+      end
+
+      def handle_async(:my_task, {:ok, fetched_org}, socket) do
+        %{org: org} = socket.assigns
+        {:noreply, assign(socket, :org, AsyncResult.ok(org, fetched_org))}
+      end
+
+      def handle_async(:my_task, {:exit, reason}, socket) do
+        %{org: org} = socket.assigns
+        {:noreply, assign(socket, :org, AsyncResult.failed(org, {:exit, reason}))}
+      end
+
+  See the moduledoc for more information.
+  """
+  def start_async(%Socket{} = socket, name, func)
+      when is_atom(name) and is_function(func, 0) do
+    Async.start_async(socket, name, func)
+  end
+
+  @doc """
+  Cancels an async operation.
+
+  Accepts either the `%AsyncResult{}` when using `assign_async/3` or
+  the keys passed to `start_async/3`.
+
+  Returns the `%Phoenix.LiveView.Socket{}`.
+
+  ## Examples
+
+      cancel_async(socket, :preview)
+      cancel_async(socket, :preview, :my_reason)
+      cancel_async(socket, [:profile, :rank])
+      cancel_async(socket, socket.assigns.preview)
+  """
+  def cancel_async(socket, async_or_keys, reason \\ :cancel) do
+    Async.cancel_async(socket, async_or_keys, reason)
+  end
+
+  @doc """
+  Cancels an async operation if one exists.
```

Comment:
What is the difference between this and the function above? Can we document what happens if you cancel_async something that does not exist? It is not immediately clear. :)

Reply:
Oh, I got it. I think we should only have `cancel_existing_async` (and it should be called `cancel_async`). Otherwise your code _will_ have race conditions. That's why Erlang doesn't raise if you give a dead pid to Process.exit and most other functions.